### ISSUE TITLE
aws_globalaccelerator_endpoint_group traffic_dial_percentage cannot be set to 0

### ISSUE THREAD

**Issue by maleko66** [terraform-providers/terraform-provider-aws#11182]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.12.8
+ provider.aws v2.41.0

### Affected Resource(s)

* aws_globalaccelerator_endpoint_group

### Terraform Configuration Files

```hcl
resource "aws_globalaccelerator_endpoint_group" "eu-west-1" {
  listener_arn            = aws_globalaccelerator_listener.some.id
  health_check_path       = ""
  health_check_port       = 443
  traffic_dial_percentage = 0
  endpoint_group_region   = "eu-west-1"

  endpoint_configuration {
    endpoint_id = "arn:some-alb-in-eu-west-1"
    weight      = 128
  }
}
```

### Expected Behavior
New endpoint with traffic dial set to 0%

### Actual Behavior
0% is ignored and new endpoint has traffic dial set to 100% (100% is the default).  In fact, 0% is always ignored.  You can't change any existing traffic_dial_percentage to 0% as this results in a no-op with subsequent `terraform plan` and `terraform apply` still wanting to change the value to 0%.

### Steps to Reproduce

1. `terraform apply`

**Comment by bflad** [comment]
Ah, thanks for reporting this @maleko66 👍 Indeed the logic in the resource would currently prevent the configuration of the value 0, because its currently using `d.GetOk("traffic_dial_percentage")`, which will return `false` if the value is 0 (matching the type's zero-value).

If you or anyone else is interesting in fixing this, we need to use `d.Get("traffic_dial_percentage").(float64)` instead, which is safe because the attribute always has a value due to the `Default: 100.0`. e.g.

```go
	opts := &globalaccelerator.CreateEndpointGroupInput{
		ListenerArn:           aws.String(d.Get("listener_arn").(string)),
		IdempotencyToken:      aws.String(resource.UniqueId()),
		EndpointGroupRegion:   aws.String(region),
		TrafficDialPercentage: aws.Float64(d.Get("traffic_dial_percentage").(float64)),
	}
// ...
	opts := &globalaccelerator.UpdateEndpointGroupInput{
		EndpointGroupArn:      aws.String(d.Id()),
		TrafficDialPercentage: aws.Float64(d.Get("traffic_dial_percentage").(float64)),
	}
```

And removing the `d.GetOk("traffic_dial_percentage")` calls. 👍

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.43.0 of the Terraform AWS Provider, tomorrow. Thanks to @zsiddique for the implementation 👍
